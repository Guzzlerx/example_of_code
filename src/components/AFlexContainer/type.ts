import { ElementType, FocusEventHandler, MouseEventHandler, ReactNode } from 'react';
import { FlattenSimpleInterpolation } from 'styled-components';

export interface IAFlexContainerProps {
  children: ReactNode;
  styles?: FlattenSimpleInterpolation;
  renderAs?: ElementType;
  className?: string;
  dataTestId?: string;
  onClick?: MouseEventHandler<HTMLElement>;
  onFocus?: FocusEventHandler<HTMLElement>;
  onBlur?: FocusEventHandler<HTMLElement>;
  isNonFlexible?: boolean;
  flex?: string;
  flexDirection?: 'row' | 'row-reverse' | 'column' | 'column-reverse';
  flexWrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
  justifyContent?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly';
  alignItems?: 'stretch' | 'flex-start' | 'flex-end' | 'center' | 'baseline';
  alignContent?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | 'stretch'
    | 'baseline';
  gap?: string;
}
