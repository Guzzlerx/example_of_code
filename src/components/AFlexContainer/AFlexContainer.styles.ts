import styled from 'styled-components';
import { IAFlexContainerProps } from './type';

export const StyledContainer = styled.div<IAFlexContainerProps>`
  display: ${({ isNonFlexible }) => (isNonFlexible ? 'block' : 'flex')};
  ${({ styles }) => styles}
  flex: ${({ flex }) => flex};
  flex-direction: ${({ flexDirection }) => flexDirection};
  flex-wrap: ${({ flexWrap }) => flexWrap};
  justify-content: ${({ justifyContent }) => justifyContent};
  align-items: ${({ alignItems }) => alignItems};
  align-content: ${({ alignContent }) => alignContent};
  gap: ${({ gap }) => gap};
`;
