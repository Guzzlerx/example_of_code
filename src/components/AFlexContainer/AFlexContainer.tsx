import { forwardRef } from 'react';
import { StyledContainer } from './AFlexContainer.styles';
import { IAFlexContainerProps } from './type';

export const AFlexContainer = forwardRef(
  (
    { renderAs = 'div', children, onFocus, onBlur, dataTestId, ...restProps }: IAFlexContainerProps,
    ref,
  ) => (
    <StyledContainer
      onFocus={onFocus}
      onBlur={onBlur}
      ref={ref}
      data-testid={dataTestId}
      as={renderAs}
      {...restProps}
    >
      {children}
    </StyledContainer>
  ),
);
