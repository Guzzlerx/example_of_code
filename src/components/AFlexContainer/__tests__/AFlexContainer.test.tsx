import { RenderResult, render } from '@testing-library/react';
import { AFlexContainer } from '../AFlexContainer';

describe('AFlexContainer', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<AFlexContainer>CHILD</AFlexContainer>);
  });

  it('should render correctly', () => {
    expect(renderResult.container).toBeInTheDocument();
  });

  it('snapshot', () => {
    expect(renderResult.container).toMatchSnapshot();
  });
});
