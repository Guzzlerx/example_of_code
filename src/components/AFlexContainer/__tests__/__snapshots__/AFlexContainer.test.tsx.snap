// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AFlexContainer snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe jVDBIf"
  >
    CHILD
  </div>
</div>
`;
